cmd/kubeless/function/deploy.go

```diff
@@ -177,11 +190,31 @@ var deployCmd = &cobra.Command{
 		}
 
 		f, err := getFunctionDescription(funcName, ns, handler, file, funcDeps, runtime, runtimeImage, mem, cpu, timeout, imagePullPolicy, port, headless, envs, labels, secrets, defaultFunctionSpec)
-
 		if err != nil {
 			logrus.Fatal(err)
 		}
 
+		if dryrun == true {
+			if output == "json" {
```

Comment:
it's better to use a switch-case here

Also I would move this block to a function that receives `format(info interface{}, format string) (string, error)` in the file `pkg/utils/kubelessutil.go` to avoid repeating the same code all the time.